### DISCUSSION TITLE
Allow actions to be hidden in the index view

### DISCUSSION THREAD

**Discussion by calebuharrison** [avo-hq/avo#575] [Feature requests]
I'd like to prevent some actions from showing up in the index view, only allowing them to appear in the show view. This could be as simple as an option set on the action class.

**Comment by calebuharrison** [comment]
Actually, now that I think about it - I'd like for this to use policies somehow. I can definitely see it being useful to give certain roles of users access to certain actions.

**Reply by calebuharrison** [reply]
Alternatively, maybe the `action` DSL could accept the `visible` option like other fields. You could also use `hide_on` or `only_on`.

**Reply by adrianthedev** [reply]
Yeah, that's what I was thinking of. Have a `visible` lambda that receives the view (and some other data) and you could toggle the visibility through that lambda.

**Reply by adrianthedev** [reply]
@nulldotpro can you have a look at [this PR](https://github.com/avo-hq/avo/pull/594)? Would this cut it?

[This](https://github.com/avo-hq/avo/pull/594/files#diff-68cf4e808fd1d81a7657f7cd38621f3aecd6412c242d0654633f07201de52749R4) is how you would use it.

**Reply by calebuharrison** [reply]
@adrianthedev Left a comment for you. Looking good, thanks so much for making this happen!
